src/AppInstallerCommonCore/Manifest/ManifestSchemaValidation.cpp

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "winget/Yaml.h"
+#include "winget/ManifestCommon.h"
+#include "winget/ManifestSchemaValidation.h"
+#include "winget/ManifestYamlParser.h"
+
+#include <ManifestSchema.h>
+
+namespace AppInstaller::Manifest::YamlParser
+{
+    using namespace std::string_view_literals;
+
+    namespace
+    {
+        enum class YamlScalarType
+        {
+            String,
+            Int
+        };
+
+        // List of fields that use non string scalar types
+        const std::map<std::string_view, YamlScalarType> ManifestFieldTypes=
+        {
+            { "InstallerSuccessCodes"sv, YamlScalarType::Int }
+        };
+
+        YamlScalarType GetManifestScalarValueType(const std::string& key)
+        {
+            auto iter = ManifestFieldTypes.find(key);
+            if (iter != ManifestFieldTypes.end())
+            {
+                return iter->second;
+            }
+
+            return YamlScalarType::String;
+        }
+
+        Json::Value YamlScalarNodeToJson(const YAML::Node& scalarNode, YamlScalarType scalarType)
+        {
+            if (scalarType == YamlScalarType::Int)
+            {
+                return Json::Value(scalarNode.as<int>());
+            }
+            else
+            {
+                return Json::Value(scalarNode.as<std::string>());
+            }
+        }
+
+        Json::Value ManifestYamlNodeToJson(const YAML::Node& rootNode, YamlScalarType scalarType = YamlScalarType::String)
+        {
+            Json::Value result;
+
+            if (rootNode.IsNull())
+            {
+                result = Json::Value::nullSingleton();
+            }
+            else if (rootNode.IsMap())
+            {
+                for (auto const& keyValuePair : rootNode.Mapping())
+                {
+                    // We only support string type as key in our manifest
+                    auto key = keyValuePair.first.as<std::string>();
+                    result[keyValuePair.first.as<std::string>()] = ManifestYamlNodeToJson(keyValuePair.second, GetManifestScalarValueType(key));
+                }
+            }
+            else if (rootNode.IsSequence())
+            {
+                for (auto const& value : rootNode.Sequence())
+                {
+                    result.append(ManifestYamlNodeToJson(value, scalarType));
+                }
+            }
+            else if (rootNode.IsScalar())
+            {
+                result = YamlScalarNodeToJson(rootNode, scalarType);
+            }
+            else
+            {
+                THROW_HR(E_UNEXPECTED);
+            }
+
+            return result;
+        }
+    }
+
+    std::string LoadResourceAsString(PCWSTR resourceModuleName, PCWSTR resourceName, PCWSTR resourceType)

```

Comment:
>PCWSTR [](start = 37, length = 6)

std::wstring_view

Reply:
Although I suppose that doesn't require null termination, so it is suspect.  We should probably add a utility type that can handle this for us.

---
In reply to: [571321711](https://github.com/microsoft/winget-cli/pull/740#discussion_r571321711) [](ancestors = 571321711)

Reply:
per offline chat, we decided to remove the resourceModuleName parameter, leave the other 2 PCWSTR as they are because that's what MAKEINTRESOURCE produces and later consumed by resource related apis

---
In reply to: [571322386](https://github.com/microsoft/winget-cli/pull/740#discussion_r571322386) [](ancestors = 571322386,571321711)